### PR TITLE
Merge release/public-v1 into develop, and CMake stuff

### PR DESCRIPTION
**Pull request by kgerheiser**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+build/
+install/
+
+*.[ao]
+*.mod
+*.so
```
Comment (aerorahul) [inline .gitignore:6]: please add `.swp`

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,31 +1,23 @@
 cmake_minimum_required(VERSION 3.15)
-project(grib_util VERSION 1.1.1)
-set(${PROJECT_NAME}_VERSION ${PROJECT_VERSION} CACHE INTERNAL "${PROJECT_NAME} version number")
-enable_language (Fortran)
+project(grib_util VERSION 1.1.1 LANGUAGES C Fortran)
```
Comment (aerorahul) [inline CMakeLists.txt:2]: we should really add a `VERSION` file.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,31 +1,23 @@
...
-set(CMAKE_FIND_PACKAGE_PREFER_CONFIG true)
+option(OPENMP "use OpenMP threading" OFF)
 
-if (NOT CMAKE_BUILD_TYPE)
-  set (CMAKE_BUILD_TYPE RELEASE CACHE STRING
-    "Choose the type of build, options are: PRODUCTION Debug Release."
-    FORCE)
-endif()
+set(CMAKE_FIND_PACKAGE_PREFER_CONFIG true)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/Modules/")
```
Comment (aerorahul) [inline CMakeLists.txt:7]: trailing `/` is unnecessary.

File: CMakeLists.txt [hunk 4]
```diff
@@ -34,42 +26,25 @@ if(POLICY CMP0012)
   cmake_policy(SET CMP0012 NEW)
 endif()
 
-find_package(OpenMP)
+if(OPENMP)
+  find_package(OpenMP)
```
Comment (aerorahul) [inline CMakeLists.txt:30]: this should be `REQUIRED` if the user wishes to turn this option. Additionally COMPONENTS should be specified.

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,31 +1,23 @@
...
 
-set(CMAKE_FIND_PACKAGE_PREFER_CONFIG true)
+option(OPENMP "use OpenMP threading" OFF)
 
-if (NOT CMAKE_BUILD_TYPE)
-  set (CMAKE_BUILD_TYPE RELEASE CACHE STRING
-    "Choose the type of build, options are: PRODUCTION Debug Release."
-    FORCE)
-endif()
+set(CMAKE_FIND_PACKAGE_PREFER_CONFIG true)
```
Comment (aerorahul) [inline CMakeLists.txt:6]: Not necessary.

**Review by BoiVuong-NOAA — APPROVED** [review]
Looks good !
Except WGRIB2, We need to have Mark's decision to remove the wgrib2's source code out of grib_util.
You can go ahead to remove the wgrib2 from grib_util

File: sorc/copygb.fd/CMakeLists.txt [hunk 6]
```diff
@@ -1,32 +1,16 @@
...
-file(GLOB fortran_src
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.f90
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.F)
+set(fortran_src copygb.F)
 
+set(exe_name copygb)
 add_executable(${exe_name} ${fortran_src})
+target_link_libraries(${exe_name} bacio::bacio_4 w3nco::w3nco_4 ip::ip_4 sp::sp_4
+  ${OpenMP_Fortran_FLAGS})
```
Comment (aerorahul) [inline sorc/copygb.fd/CMakeLists.txt:14]: use `OpenMP::OpenMP_Fortran`
Also,  wrap this in an if block of  `OpenMP_Fortran_FOUND`

File: sorc/copygb2.fd/CMakeLists.txt [hunk 7]
```diff
@@ -1,36 +1,35 @@
...
+target_link_libraries(
+  ${exe_name}
+  g2::g2_d
+  PNG::PNG
+  ${JASPER_LIBRARIES}
+  bacio::bacio_4
+  w3nco::w3nco_d
+  ip::ip_d
+  sp::sp_d
+  ${OpenMP_Fortran_FLAGS})
```
Comment (aerorahul) [inline sorc/copygb2.fd/CMakeLists.txt:33]: Same comment as `copygb.fd` on `OpenMP`. See above.

File: sorc/degrib2.fd/CMakeLists.txt [hunk 8]
```diff
@@ -1,34 +1,22 @@
-
-if(IntelComp)
-  set(fortran_flags "-g" "-O3" "-convert" "big_endian" "-axCORE-AVX2")
-elseif(GNUComp)
-  set(fortran_flags "-g" "-O3" "-fconvert=big-endian")
-else()
-  message("unknown compiler!")
-  exit()
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -convert big_endian -axCORE-AVX2 -fpp")
```
Comment (aerorahul) [inline sorc/degrib2.fd/CMakeLists.txt:2]: axCORE-AVX2 is useless on some machines with Intel. But OK!

**Review by aerorahul — CHANGES_REQUESTED** [review]
see comments inline.

**Comment by aerorahul** [comment]
@kgerheiser is this PR still being worked on?

**Comment by BoiVuong-NOAA** [comment]
Yes, The PR still works

File: sorc/degrib2.fd/CMakeLists.txt (hunk 8 above)
Comment (BoiVuong-NOAA) [reply sorc/degrib2.fd/CMakeLists.txt:2]: You are correct.  We still need it on WCOSS.

**Pull request closed without merging**